context/useSessionContext.tsx

```diff
@@ -99,56 +73,71 @@ export const SessionProvider = ({ children }: any) => {
   };
 
   useEffect(() => {
-    if (
-      lastUsedTrainingSettings?.repo !== repo?.repo ||
-      lastUsedTrainingSettings?.owner !== repo?.owner ||
-      lastUsedTrainingSettings?.lofaf.length !== lofaf.length
-    ) {
-      setMessages([]); //reset messages
-      setupContextMessages();
-    }
-  }, [lofaf, repo, session, user, repo?.repo]);
-
-  useEffect(() => {
-    //set user and session
-    if (supabase) {
-      const setData = async () => {
-        if (supabase) {
+    if (isMockEnabled) {
+      setUser(mockManager.mockData().user as unknown as User);
+      setSession(mockManager.mockData().session as unknown as Session);
+      setIsPro(mockManager.mockData().isPro);
+      setRepo(mockManager.mockData().repo);
+      setLofaf(mockManager.mockData().lofaf);
+      setTechStack(mockManager.mockData().techStack);
+      setContext(mockManager.mockData().context);
+      setBranch(mockManager.mockData().branch);
+      setMessages(mockManager.mockData().messages);
+    } else {
+  
+      if (supabase) {
+        const setData = async () => {
           const {
             data: { session },
             error,
+            // @ts-ignore
           } = await supabase.auth.getSession();
           if (error) throw error;
           setSession(session);
-          setUser(session?.user);
-        }
-      };
+          // @ts-ignore
+          setUser(session.user);
```

Comment:
This seems to be causing occasional errors, might need changing to session?.user